bot/exts/events/advent_of_code/_cog.py

```diff
@@ -96,7 +96,9 @@ async def completionist_task(self) -> None:
                 # Only give the role to people who have completed all 50 stars
                 continue
 
-            member_id = aoc_name_to_member_id.get(member_aoc_info["name"], None)
+            aoc_name = member_aoc_info["name"] if member_aoc_info["name"] else f"Anonymous #{member_aoc_info['id']}"
```

Comment:
This is more of a nitpick, but this can be simplified.
```suggestion
            aoc_name = member_aoc_info["name"] or f"Anonymous #{member_aoc_info['id']}"
```